File: ABC083B.ml
(* ABC083B - Some Sums *)

let rec sum_of_digits x =
  if x >= 10 then (x mod 10) + sum_of_digits (x / 10) else x

let solver = function
  | [ n; a; b ] ->
      let test x = a <= x && x <= b in
      let summary = ref 0 in
      for n' = 1 to n do
        if test (sum_of_digits n') then summary := !summary + n'
      done;
      !summary
  | _ -> failwith "?"

let () =
  read_line ()
  |> Str.split (Str.regexp " ")
  |> List.map int_of_string |> solver |> string_of_int |> print_endline

File: ABC083B.cpp
// ABC083B - Some Sums
#include <cstdlib>
#include <iostream>

long sum_of_digits(long x)
{
    auto y = 0;
    while (x > 0)
    {
        auto n = std::div(x, 10L);
        x = n.quot;
        y = y + n.rem;
    }
    return y;
}

long solver(long N, long A, long B)
{
    auto summary = 0L;
    for (auto i = 1L; i <= N; ++i)
    {
        if (auto s = sum_of_digits(i); A <= s && s <= B)
        {
            summary += i;
        }
    }
    return summary;
}

int main()
{
    long N, A, B;
    std::cin >> N >> A >> B;
    std::cout << solver(N, A, B) << std::endl;
    return 0;
}
